### PR TITLE
Victor/move dfu over bm_core

### PR DESCRIPTION
**Pull request by victorsowa12**
Copy over the dfu files from bm_protocol and update them with the latest style guides.

### CODE REVIEW

**Review by matt001k — APPROVED** [review]
Looks nice. A couple of nitpicks on code styling.

File: include/dfu/dfu.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  uint8_t minor;
+  uint64_t host_node_id;
+  uint32_t gitSHA;
+} ReboootClientUpdateInfo;
+
+#ifndef CI_TEST
+extern ReboootClientUpdateInfo client_update_reboot_info __attribute__((section(".noinit")));
+#else // CI_TEST
+extern ReboootClientUpdateInfo client_update_reboot_info;
+#endif // CI_TEST
```
[the remark this answers is not in the document]
Comment (victorsowa12) [reply include/dfu/dfu.h:102]: It was.. I just copied it over. Not sure if we will still need it or not.

File: include/dfu/dfu.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#ifndef __BM_DFU_H__
+#define __BM_DFU_H__
```
[the remark this answers is not in the document]
Comment (victorsowa12) [reply include/dfu/dfu.h:2]: Yeah both `dfu_client.h` and `dfu_host.h` require the `BcmpDfuTxFunc` defined in `dfu.h`

File: include/dfu/dfu.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorsowa12) [reply include/dfu/dfu.h:103]: I just added a todo for it for now

**Pull request merged**